src/Invictus.Testing/LogicAppsProvider.cs

```diff
@@ -0,0 +1,307 @@
+            ILogger logger)
+        {
+            Guard.NotNullOrWhitespace(resourceGroup, nameof(resourceGroup));
+            Guard.NotNullOrWhitespace(logicAppName, nameof(logicAppName));
+            Guard.NotNull(authentication, nameof(authentication));
+            
+            logger = logger ?? NullLogger.Instance;
+            return new LogicAppsProvider(resourceGroup, logicAppName, authentication, logger);
+        }
+
+        /// <summary>
+        /// Sets the start time when the logic app runs were executed.
+        /// </summary>
+        /// <param name="startTime">The date that the logic app ran.</param>
+        public LogicAppsProvider WithStartTime(DateTime startTime)
+        {
+            // TODO: just migrated from original 'helper' class, should be with offset?
+            _startTime = startTime;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the time period in which the retrieval of the logic app runs should succeed.
+        /// </summary>
+        /// <param name="timeout">The period to retrieve logic app runs.</param>
+        public LogicAppsProvider WithTimeout(TimeSpan timeout)
+        {
+            _timeout = timeout;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the correlation ID as client tracking when retrieving logic app runs.
+        /// </summary>
+        /// <param name="correlationId">The client tracking of the logic app runs.</param>
+        public LogicAppsProvider WithCorrelationId(string correlationId)
+        {
+            Guard.NotNull(correlationId, nameof(correlationId));
+
+            _hasCorrelationId = true;
+            _correlationId = correlationId;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the tracking property as filter when retrieving logic app runs.
+        /// </summary>
+        /// <param name="trackedPropertyName">The name of the tracked property to filter on.</param>
+        /// <param name="trackedPropertyValue">The value of the tracked property to filter on.</param>
+        public LogicAppsProvider WithTrackedProperty(string trackedPropertyName, string trackedPropertyValue)
+        {
+            Guard.NotNull(trackedPropertyName, nameof(trackedPropertyName));
+            Guard.NotNull(trackedPropertyValue, nameof(trackedPropertyValue));
+
+            _hasTrackedProperty = true;
+            _trackedPropertyName = trackedPropertyName;
+            _trackedPropertyValue = trackedPropertyValue;
+            return this;
+        }
+
+        /// <summary>
+        /// Start polling for a single logic app run corresponding to the previously set filtering criteria.
+        /// </summary>
+        public async Task<LogicAppRun> PollForSingleLogicAppRunAsync()
+        {
+            IEnumerable<LogicAppRun> logicAppRuns = await PollForLogicAppRunsAsync(numberOfItems: 1);
+            return logicAppRuns.FirstOrDefault();
```

Comment:
Let's use this to be :feelsgood: 
```suggestion
            return logicAppRuns.SingleOrDefault();
```